ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -1305,7 +1322,9 @@ public static Directory getAcidState(FileSystem fileSystem, Path candidateDirect
             bestBase, ignoreEmptyFiles, abortedDirectories, fs, validTxnList);
       }
     } else {
-      dirSnapshots = getHdfsDirSnapshots(fs, candidateDirectory);
+      if (dirSnapshots == null) {
```

Comment:
There is a slight problem here, if we are on hdfs and the file listing with id is supported. Few lines below there is a check for dirsnapshot == null, that was running every time for this case, but now it won't run if you call getacidstate with nonnull dirsnapshot